sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1224,6 +1224,16 @@ object SQLConf {
     .stringConf
     .createOptional
 
+  val FILE_STAGING_DIR =
+    buildConf("spark.sql.source.stagingDir")
```

Comment:
This looks orthogonal what your PR aims. Could you spin off this new configuration as a new PR?

Reply:
Let me do this first.